File: Day 2/Day2.java
import java.util.HashMap;
import java.util.Scanner;

public class Day2 {

    public static boolean is_possible(String[] sets) {
        int red_shown = 0;
        int green_shown = 0;
        int blue_shown = 0;

        for(String game : sets){
            String[] games = game.split(",");

            for(int i = 0; i < games.length; i++){
                
            }
        }
        return true;
    }
    public static void main(String[] args) {
        int red_total = 12;
        int green_total = 13;
        int blue_total = 14;

        HashMap<String,Integer> totals = new HashMap<>();
        totals.put("red", 12);
        totals.put("green", 13);
        totals.put("blue", 14);

        Scanner scan = new Scanner("'Day 2\\sample.txt'");

        while(scan.hasNextLine()){
            System.out.println(scan.nextLine());
        }


        scan.close();
    }
}

File: Day 2/day2.py
import re


red_total = 12
green_total = 13
blue_total = 14

totals = {
    'red': 12,
    'green': 13,
    'blue': 14
}

def is_possible(sets: list[str]):
    red_shown = 0
    green_shown = 0
    blue_shown = 0
    for game in sets:
        for curr_game in game.split(','):
            curr_game = curr_game.strip()
            if (re.findall('red', curr_game)):
                red_shown += int(curr_game[0, curr_game.find(' ')])
            elif (re.findall('green', curr_game)):
                green_shown += int(curr_game[0, curr_game.find(' ')])
            elif (re.findall('blue', curr_game)):
                blue_shown += int(curr_game[0, curr_game.find(' ')])
            
        print(f'Red: {red_shown}\tGreen: {green_shown}\tBlue: {blue_shown}\n')
        if ((red_shown > red_total) or 
            (green_shown > green_total) or
            (blue_shown > blue_total)):
                return False
    return True

with open(r'Day 2\sample.txt') as input:
    for line in input:
        # First grab the id of the game
        groups = re.split(pattern=':', string=line)
        id = groups[0][-1]
        
        sets = groups[1].replace('\n', '').split(';')
        print(is_possible(sets))
        print('===============')
